index.bs

```diff
@@ -5523,6 +5529,126 @@ opened</dfn> steps given |window|, |type|, |message|, and optional |default valu
+
+The <dfn export for=commands>emulation.setGeolocationOverride</dfn> command modifies geolocation characteristics on the given top-level traversables or user contexts.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      emulation.SetGeolocationOverride = (
+        method: "emulation.setGeolocationOverride",
+        params: emulation.SetGeolocationOverrideParameters
+      )
+
+      emulation.SetGeolocationOverrideParameters = {
+        coordinates: emulation.GeolocationCoordinates / null,
+        ? contexts: [+browsingContext.BrowsingContext],
+        ? userContexts: [+browser.UserContext],
+      }
+
+      emulation.GeolocationCoordinates = {
+         latitude: float,
+         longitude: float,
+         ? accuracy: float .default 1.0,
+         ? altitude: float / null .default null,
+         ? altitudeAccuracy: float / null .default null,
+         ? heading: float / null .default null,
+         ? speed: float / null .default null,
+      }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <code>
+     EmptyResult
+    </code>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for emulation.setGeolocationOverride">
+
+The [=remote end steps=] with |command parameters| are:
+
+1. If |command parameters| [=map/contains=] "<code>userContexts</code>"
+   and |command parameters| [=map/contains=] "<code>context</code>",
+   return [=error=] with [=error code=] [=invalid argument=].
+
+1. Let |navigables| be a [=/set=].
+
+1. If the <code>contexts</code> field of |command parameters| is present:
+
+   1. For each |context| of |command parameters|["<code>contexts</code>"]:
+
+      1. Let |navigable| be the result of [=trying=] to [=get a navigable=] with
+         |context|.
+
+      1. If |navigable| is not a [=/top-level traversable=], return [=error=] with
+         [=error code=] [=invalid argument=].
+
+      1. [=set/Append=] |navigable| to |navigables|.
+
+1. Otherwise, if the <code>userContexts</code> field of |command parameters| is present:
+
+   1. Let |user contexts| be a [=/set=].
+
+   1. For each |user context id| of |command parameters|["<code>userContexts</code>"]:
+
+      1. Set |user context| to [=get user context=] with |user context id|.
+
+      1. If |user context| is null, return [=error=] with [=error code=] [=no such user context=].
+
+      1. [=set/Append=] |user context| to |user contexts|.
+
+   1. For each |user context| of |user contexts|:
+
+      1. [=map/Set=] [=geolocation overrides map=][|user context|] to a |command parameters|["<code>coordinates</code>"].
+
+      1. [=list/For each=] |top-level traversable| of the list of all [=/top-level traversables=]
+         whose [=associated user context=] is |user context|:
+
+         1. [=list/Append=] |top-level traversable| to |navigables|.
+
+1. Otherwise, return [=error=] with [=error code=] [=invalid argument=].
```

Comment:
I don't like the multiple-otherwise formulation ("otherwise" is already confusing because it's not clear if it should be a separate step or not). I'd rather have this failure at the top.

Reply:
yeah, I agree, updated